### PR TITLE
Add a section the SnowflakeConnection API Reference on configuration

### PR DESCRIPTION
**Pull request by sfc-gh-jcarroll**
## 📚 Context

Provide more dedicated and detailed documentation about configuring a SnowflakeConnection. Specifically, want to deep link this from an error in the library when a developer doesn't have any config available.

## 🧠 Description of Changes

Added the "Configuration" section to [SnowflakeConnection API Reference](https://docs.streamlit.io/library/api-reference/connections/st.connections.snowflakeconnection)

## 💥 Impact

Modest - new section with clearer documentation about options available and links to relevant Snowflake docs for more information.

**[Updated section in Deploy Preview](https://deploy-preview-873--streamlit-docs.netlify.app/library/api-reference/connections/st.connections.snowflakeconnection#configuration)**

Size:

- [ ] Small 
- [x] Not small 

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

File: content/library/api/connections/connections-snowflake.md
Comment (sfc-gh-dmatthews) [inline content/library/api/connections/connections-snowflake.md:21]: > It can also use existing Snowflake connection configuration when available.

This feels a tiny bit cryptic, but maybe I'm overthinking it since anyone using SnowflakeConnection should know what Snowflake and presumably SiS is.

I know it's explained further on, but to avoid a momentary pause of confusion, would it make sense to be a bit more concrete here?
> It can also use an existing connection configuration when running within Snowflake.

**Review by sfc-gh-dmatthews — APPROVED** [review]

File: content/library/api/connections/connections-snowflake.md
Comment (sfc-gh-jcarroll) [reply content/library/api/connections/connections-snowflake.md:21]: Want to make sure we distinguish these two different scenarios (your last quote there seems to confuse them):
- When running in Snowflake (SIS), no configuration is needed. the credentials are provided by context.
- When running outside Snowflake, the configuration like connections.toml can be used

If it's not clear enough, could we either link to the anchor below or link [here again](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-example#connecting-using-the-connections-toml-file)?

File: content/library/api/connections/connections-snowflake.md
Comment (sfc-gh-jcarroll) [inline content/library/api/connections/connections-snowflake.md:14]: 100% your call @sfc-gh-dmatthews but FWIW I had this at H3 originally and felt like it looked better at H2

image

Comment (sfc-gh-dmatthews) [reply content/library/api/connections/connections-snowflake.md:14]: I can be convinced either way. I agree the H2+H3 is prettier than the H3+H4, but logically it's a child to the class which is hard-coded as H2. I mentioned this in Slack, but I want to make some changes to the header sizing and be able to make a class H1 with its methods as H2.

Happy to go either way and address it later for larger consistency.

Comment (sfc-gh-jcarroll) [reply content/library/api/connections/connections-snowflake.md:14]: OK. Yeah I think it should eventually be the same level as the methods. Whatever is easier for you.

Comment (sfc-gh-jcarroll) [reply content/library/api/connections/connections-snowflake.md:14]: I'm good to merge it like this if you are going to fix the consistency later :)

**Pull request merged**
